File: Android/app/src/main/java/com/example/snaptaplaque/models/api/account/MeResponse.java
package com.example.snaptaplaque.models.api.account;

/**
 * Modèle de données représentant les informations détaillées du profil de l'utilisateur connecté.
 * <p>Cette classe est généralement utilisée pour réceptionner les données de l'endpoint "/me".
 * Elle contient l'identité de l'utilisateur, son statut d'activité, ses privilèges d'administration
 * ainsi que les métadonnées de création de son compte.</p>
 */
public class MeResponse {
    /** Identifiant unique de l'utilisateur dans la base de données. */
    private Integer id;
    /** Adresse e-mail associée au compte utilisateur. */
    private String email;
    /** Nom d'utilisateur unique utilisé pour l'identification. */
    private String username;
    /** Nom complet (Prénom et Nom) de l'utilisateur. */
    private String full_name;
    /** Indique si le compte est actuellement actif et autorisé à se connecter. */
    private Boolean is_active;
    /** Indique si l'utilisateur possède les droits d'administration sur la plateforme. */
    private Boolean is_admin;
    /** Date et heure de création du compte (généralement au format ISO 8601). */
    private String created_at;

    /**
     * Constructeur complet pour initialiser les informations de profil.
     *
     * @param id         L'identifiant unique.
     * @param email      L'adresse e-mail.
     * @param username   Le nom d'utilisateur.
     * @param full_name  Le nom complet.
     * @param is_active  Le statut d'activité du compte.
     * @param is_admin   Le statut administrateur.
     * @param created_at La date de création du compte.
     */
    public MeResponse(Integer id, String email, String username, String full_name, Boolean is_active, Boolean is_admin, String created_at) {
        this.id = id;
        this.email = email;
        this.username = username;
        this.full_name = full_name;
        this.is_active = is_active;
        this.is_admin = is_admin;
        this.created_at = created_at;
    }

    /**
     * Récupère l'identifiant unique de l'utilisateur.
     *
     * @return L'ID sous forme d'{@link Integer}.
     */
    public Integer getId() {
        return id;
    }

    /**
     * Récupère l'adresse e-mail de l'utilisateur.
     *
     * @return L'adresse e-mail sous forme de {@link String}.
     */
    public String getEmail() {
        return email;
    }

    /**
     * Récupère le nom d'utilisateur.
     *
     * @return Le pseudo sous forme de {@link String}.
     */
    public String getUsername() {
        return username;
    }

    /**
     * Récupère le nom complet de l'utilisateur.
     *
     * @return Le nom complet sous forme de {@link String}.
     */
    public String getFull_name() {
        return full_name;
    }

    /**
     * Vérifie si le compte est actif.
     *
     * @return {@code true} si le compte est actif, {@code false} sinon.
     */
    public Boolean getIs_active() {
        return is_active;
    }

    /**
     * Vérifie si l'utilisateur possède les droits d'administration.
     *
     * @return {@code true} si l'utilisateur est admin, {@code false} sinon.
     */
    public Boolean getIs_admin() {
        return is_admin;
    }

    /**
     * Récupère la date de création du compte.
     *
     * @return La date de création sous forme de {@link String}.
     */
    public String getCreated_at() {
        return created_at;
    }
}

File: iOS/SnapTaPlaque/Models/API/Account/MeResponse.swift
//
//  MeResponse.swift
//  SnapTaPlaque
//
//  Created by Victor Poirier on 20/03/2026.
//

import Foundation

/// Représente les données de profil de l'utilisateur connecté ("Me").
///
/// Ce modèle est utilisé par l'application locale pour parser la réponse de l'API (`v1/account/me`)
/// et construire le compte utilisateur dans l'interface, conformément aux conventions REST.
/// Il implémente `Codable` pour la désérialisation du JSON et `Identifiable`
/// pour faciliter l'utilisation directe dans les listes `SwiftUI`.
struct MeResponse: Codable, Identifiable {
    
    /// Un identifiant unique correspondant à l'ID en base de données.
    let id: Int
    
    /// L'adresse courriel (e-mail) attachée au profil.
    let email: String
    
    /// Le nom d'utilisateur (pseudo) choisi lors de l'inscription pour se connecter.
    let username: String
    
    /// Le nom complet de l'utilisateur, qui sera affiché en évidence sur son profil.
    let fullName: String
    
    /// Indique si le compte de cet utilisateur est actuellement actif sur la plateforme.
    ///
    /// Utilisé potentiellement pour masquer des interfaces ou forcer une reconnexion
    /// en cas de suspension ou clôture depuis l'administration.
    let isActive: Bool
    
    /// Indique si cet utilisateur dispose des droits "Administrateur".
    let isAdmin: Bool
    
    /// La date de création du compte, retournée sous forme de chaîne de caractères par l'API (ISO 8601).
    let createdAt: String
    
    /// Associe les propriétés de la structure aux clés exactes du JSON renvoyé par le backend.
    ///
    /// Gère la conversion des standards de nommage de l'API (généralement `snake_case`)
    /// vers les conventions de nommage locale Swift (`camelCase`).
    enum CodingKeys: String, CodingKey {
        case id, email, username
        
        /// Mappe `fullName` depuis la clé `full_name`.
        case fullName = "full_name"
        
        /// Mappe `isActive` depuis la clé `is_active`.
        case isActive = "is_active"
        
        /// Mappe `isAdmin` depuis la clé `is_admin`.
        case isAdmin = "is_admin"
        
        /// Mappe `createdAt` depuis la clé `created_at`.
        case createdAt = "created_at"
    }
}
